go/libraries/doltcore/schema/schema.go

```diff
@@ -26,16 +26,34 @@ import (
 	"github.com/dolthub/dolt/go/store/val"
 )
 
-// Schema is an interface for retrieving the columns that make up a schema
+// Schema defines the schema of a table and describes both its SQL schema and storage layout.
+//
+// For example, a SQL table defined as:
+//
+//	`CREATE TABLE t (a int, b int, pk2 int, c int, pk1 int, PRIMARY KEY (pk1, pk2));`
+//
+// Has a corresponding Schema of:
+//
+//	Schema {
+//		PkCols:     [pk1, pk2],
+//		NonPkCols:  [a, b, c],
+//		AllCols:    [a, b, pk2, c, pk1],
+//		PkOrdinals: [4, 2],
+//	}
 type Schema interface {
-	// GetPKCols gets the collection of columns which make the primary key. They
-	// are always returned in ordinal order.
+	// GetPKCols gets the collection of columns which make the primary key.
+	// Columns in this collection are ordered by storage order, which is
+	// defined in the 'PRIMARY KEY(...)' clause of a CREATE TABLE statement.
 	GetPKCols() *ColCollection
 
 	// GetNonPKCols gets the collection of columns which are not part of the primary key.
+	// Columns in this collection are ordered by schema order (display order), which is
```

Comment:
are there instances where nonPkCols schema and storage order is not the same?

Reply:
currently, we always store them in schema order. In the future we'd like to reorder them to place fixed-width columns first to enable fast-access optimizaitons